purple-pill/js/controllers/fontexportController.js

```diff
@@ -20,152 +92,136 @@ function($scope, $http, sharedScope, $timeout) {
         $scope.data.localmenu.fonts = false;
     };
 
-    var export_is_running = false
-      , generator
-      , io = {}
-      ;
-    $scope.data.exportFonts = function() {
-
-        //Do not trigger the export routine if no instance is selected for export,
-        //otherwise it would result in an empty ZIP file.
-        if (!$scope.data.instancesForExport())
-            return;
-
-        //Also avoid triggering a new export while we're still not finished
-        if (export_is_running)
-            return;
-
-        export_is_running = true;
-        resetProgressBar();
-
-        function zero_padding(value){
+    function getTimestamp(){
+        var year, month, day, hours, minutes, seconds
+          , date = new Date()
+          ;
+        function zeroPadding(value){
             return value < 10 ? "0" + String(value) : String(value);
         }
+        year = zeroPadding(date.getFullYear());
+        month = zeroPadding(date.getMonth() + 1); //getMonth() returns a zero-based value: 0=January, 11=December
+        day = zeroPadding(date.getDate());
+        hours = zeroPadding(date.getHours());
+        minutes = zeroPadding(date.getMinutes());
+        seconds = zeroPadding(date.getSeconds());
+
+        return [year, month, day].join("") + "-" + [hours, minutes, seconds].join("");
+    }
+
+    const UI_UPDATE_TIMESLICE = 50; // msecs
+    var exportIsRunning = false;
+
+    function exportingGlyphMessage (it, instanceIndex, totalInstances){
+        var msg
+          , currentGlyph = it.value['current_glyph'] + 1 //humans start counting from 1.
+          , totalGlyphs = it.value['total_glyphs']
+          , glyphId = it.value['glyph_id']
+          ;
+        msg = totalInstances + " instances to export."
+        msg += "Calculating glyph '" + glyphId + "' (" + currentGlyph + " of " + totalGlyphs + ")"
+        msg += " of instance #" + (instanceIndex+1);
+        return msg;
+    }
+
+    function calculateGlyphsProgress(it, instanceIndex, totalInstances){
```

Comment:
The first argument should become `data` and it will be called like this
`calculateGlyphsProgress (it.value, instanceIndex, totalInstances)`
This method should have to know about iterator return objects, it works just on the data.